Installation issue: perl-xml-libxml fails in gcc14

### Steps to reproduce the issue

```console
$ spack install perl-xml-libxml@2.0201%gcc@14
...
```

### Error message

<details><summary>Error message</summary>
The `perl-xml-libxml` package fails to build with gcc14 at version 2.0201.
<pre>
...
    67    perl-libxml-mm.c: In function 'PmmDumpRegistry':
  >> 68    perl-libxml-mm.c:142:32: error: passing argument 2 of 'xmlHashScan' from incompatible pointer type [-Wincompatible-pointer-types]
     69      142 |                 xmlHashScan(r, PmmRegistryDumpHashScanner, NULL);
     70          |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
     71          |                                |
     72          |                                void (*)(void *, void *, xmlChar *) {aka void (*)(void *, void *, unsigned char *)}
     73    In file included from /home/teague/System/apps/spack/opt/spack/linux-pop22-zen3/gcc-14.1.0/libxml2-2.10.3-4zgiwk2ttsnsgbaayg7c7bjxv2ibp4mg/include/libxml2/libxml/parser.h:16,
     74                     from perl-libxml-mm.h:20,
     75                     from perl-libxml-mm.c:26:
     76    /home/teague/System/apps/spack/opt/spack/linux-pop22-zen3/gcc-14.1.0/libxml2-2.10.3-4zgiwk2ttsnsgbaayg7c7bjxv2ibp4mg/include/libxml2/libxml/hash.h:213:57: note: expected 'xmlHashScanner' {aka 'void (
           *)(void *, void *, const unsigned char *)'} but argument is of type 'void (*)(void *, void *, xmlChar *)' {aka 'void (*)(void *, void *, unsigned char *)'}
     77      213 |                                          xmlHashScanner f,
     78          |                                          ~~~~~~~~~~~~~~~^
     79    perl-libxml-mm.c: In function 'PmmUnregisterProxyNode':
...
</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (bd8a6a9925e10e34c2325d5a92c867fd2fdd036e)
* **Python:** 3.10.12
* **Platform:** linux-pop22-zen3
* **Concretizer:** clingo


### Additional information

I will be adding a `conflicts` and new `version` in  pull request in to triage this.
[spack-build-out.txt](https://github.com/spack/spack/files/15359934/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/15359936/spack-build-env.txt)

https://lists.opensuse.org/archives/list/bugs@lists.opensuse.org/thread/R4MJ6PJCJ6QUIA256UTFEBYZOCBDIZ53/?sort=thread

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
+++ b/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
@@ -19,10 +19,14 @@ class PerlXmlLibxml(PerlPackage):
 
     license("GPL-1.0-or-later OR Artistic-1.0-Perl")
 
+    version("2.0210", sha256="a29bf3f00ab9c9ee04218154e0afc8f799bf23674eb99c1a9ed4de1f4059a48d")
+    version("2.0209", sha256="b4a5abbcd689aa2fbbc8b7b45339e961c4984e48108494eb6c282b4748222425")
     version("2.0201", sha256="e008700732502b3f1f0890696ec6e2dc70abf526cd710efd9ab7675cae199bc2")
+
 
     depends_on("libxml2")
     depends_on("perl-xml-namespacesupport", type=("build", "run"))
     depends_on("perl-xml-sax", type=("build", "run"))
     depends_on("perl-xml-sax-base", type=("build", "run"))
     depends_on("perl-alien-libxml2", type="build")
+    conflicts("%gcc@14", when="@:2.0209")

--- a/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
+++ b/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
@@ -28,4 +28,5 @@ class PerlXmlLibxml(PerlPackage):
     depends_on("perl-xml-sax", type=("build", "run"))
     depends_on("perl-xml-sax-base", type=("build", "run"))
     depends_on("perl-alien-libxml2", type="build")
-    conflicts("%gcc@14", when="@:2.0209")
+
+    conflicts("%gcc@14:", when="@:2.0209")

--- a/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
+++ b/var/spack/repos/builtin/packages/perl-xml-libxml/package.py
@@ -23,7 +23,6 @@ class PerlXmlLibxml(PerlPackage):
     version("2.0209", sha256="b4a5abbcd689aa2fbbc8b7b45339e961c4984e48108494eb6c282b4748222425")
     version("2.0201", sha256="e008700732502b3f1f0890696ec6e2dc70abf526cd710efd9ab7675cae199bc2")
 
-
     depends_on("libxml2")
     depends_on("perl-xml-namespacesupport", type=("build", "run"))
     depends_on("perl-xml-sax", type=("build", "run"))

